core/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -296,6 +293,11 @@ public String toString() {
         }
     }
 
+    public boolean supersedes(ClusterState other) {
```

Comment:
And even better would be a static method for superseding that could be unit tested